frontend/javascripts/oxalis/view/components/command_palette.tsx

```diff
@@ -0,0 +1,198 @@
+      };
+    });
+};
+
+const getLabelForPath = (key: string) =>
+  getPhraseFromCamelCaseString(capitalize(key.split("/")[1])) || key;
+
+export const CommandPalette = ({ label }: { label: string | null }) => {
+  const userConfig = useSelector((state: OxalisState) => state.userConfiguration);
+  const isViewMode = useSelector(
+    (state: OxalisState) => state.temporaryConfiguration.controlMode === "VIEW",
+  );
+  const isInTracingView = useSelector(
+    (state: OxalisState) => state.uiInformation.isInAnnotationView,
+  );
+
+  const props: TracingViewMenuProps = useSelector((state: OxalisState) => {
+    return {
+      restrictions: state.tracing.restrictions,
+      task: state.task,
+      annotationType: state.tracing.annotationType,
+      annotationId: state.tracing.annotationId,
+      activeUser: state.activeUser,
+      isAnnotationLockedByUser: state.tracing.isLockedByOwner,
+      annotationOwner: state.tracing.owner,
+      isDownloadModalOpen: state.uiInformation.showDownloadModal,
+      isRenderAnimationModalOpen: state.uiInformation.showRenderAnimationModal,
+      isShareModalOpen: state.uiInformation.showShareModal,
+      isMergeModalOpen: state.uiInformation.showMergeAnnotationModal,
+      isZarrPrivateLinksModalOpen: state.uiInformation.showZarrPrivateLinksModal,
+      isUserScriptsModalOpen: state.uiInformation.showAddScriptModal,
+    };
+  });
+
+  const { activeUser, restrictions } = props;
+
+  const getMenuActions = (isViewMode: boolean) => {
+    if (!isInTracingView) return [];
+    if (isViewMode) {
+      const { items } = getViewDatasetMenu(null);
+      return items;
+    }
+    const { menuItems } = getTracingViewModalsAndMenuItems(props, null);
+    return menuItems;
+  };
+
+  const getTabsAndSettingsMenuItems = () => {
+    if (!isInTracingView) return [];
+    const commands: Command[] = [];
+
+    (Object.keys(userConfig) as [keyof UserConfiguration]).forEach((key, counter) => {
+      if (typeof userConfig[key] === "boolean") {
+        commands.push({
+          id: counter,
+          name: `Toggle ${getPhraseFromCamelCaseString(key)}`,
+          command: () => Store.dispatch(updateUserSettingAction(key, !userConfig[key])),
+          color: "#5660ff",
+        });
+      }
+    });
+    return commands;
+  };
+
+  const getNavigationEntries = () => {
+    if (activeUser == null) return [];
+    const commands: Command[] = [];
+    const basicNavigationEntries = [
+      { name: "Tasks (Dashboard)", path: "/dashboard/tasks" },
+      { name: "Annotations", path: "/dashboard/annotations" },
+      { name: "Datasets", path: "/dashboard/datasets" },
+      { name: "Time Tracking", path: "/timetracking" },
+    ];
+
+    const adminMenu = getAdministrationSubMenu(false, activeUser);
+    const adminCommands: Array<{ name: string; path: string }> = [];
+    adminMenu?.children.map((entry: { key: string }) => {
+      adminCommands.push({ name: getLabelForPath(entry.key), path: entry.key });
+    });
```

Comment:
a map with side effects is not idiomatic. how about:


```suggestion
    const adminCommands = adminMenu == null ? [] : adminMenu.children.map((entry: { key: string }) => 
      ({ name: getLabelForPath(entry.key), path: entry.key })
    );
```